test/IRGen/archetype_resilience.sil

```diff
@@ -0,0 +1,27 @@
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) %s -enable-resilience -emit-ir | %FileCheck %s --check-prefix=CHECK
+// REQUIRES: objc_interop
+
+sil_stage canonical
+
+import Swift
+import Foundation
+
+public protocol ProtArchClass: class {
+}
+
+public enum EnumWithClassArchetypeAndDynamicSize<T : ProtArchClass> {
```

Comment:
Does 'T : AnyObject' also expose the bug, without having to define a new protocol?

Reply:
It should. thought this made the example clearer..

Reply:
Ok, but ProtArchClass is a misleading name and doesn't convey any additional info. Anyobject would be better

Reply:
Updated the PR / test case to use AnyObject.